python/GafferUI/ColorSwatchPlugValueWidget.py

```diff
@@ -214,3 +252,34 @@ def __buttonClicked( self, button ) :
 	def __destroy( self, *unused ) :
 
 		self.parent().removeChild( self )
+
+	# \todo Extract these two methods to share with `ColorChooserPlugValueWidget` which has
+	# an almost identical implementation.
+
+	def __colorChooserOptionChanged( self, colorChooser, value, key ) :
+
+		if Gaffer.Metadata.value( "colorChooser:dialogueOptions", "userDefault" ) is None :
+			sessionOptions = Gaffer.Metadata.value( "colorChooser:dialogueOptions", "sessionDefault" )
+			if sessionOptions is None :
+				sessionOptions = IECore.CompoundData()
```

Comment:
Smooshing all the options into a single CompoundData isn't ideal, because it makes it impossible for someone to override just one thing in isolation, which thwarts the cascading nature of startup files. Metadata is already a key-value store, so I think we can just use multiple keys instead of a different dictionary.

Reply:
I got a little overzealous trying to reduce metadata entries. Fixed in TGCCMO2024: https://github.com/GafferHQ/gaffer/pull/6003/commits/8bf9543bafcf26dd41e265f7f18376a2d8460a4a